hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQL10Dialect.java

```diff
@@ -6,14 +6,20 @@
  */
 package org.hibernate.dialect;
 
+import org.hibernate.dialect.function.StandardSQLFunction;
 import org.hibernate.dialect.identity.IdentityColumnSupport;
 import org.hibernate.dialect.identity.PostgreSQL10IdentityColumnSupport;
+import org.hibernate.type.StandardBasicTypes;
 
 /**
  * An SQL dialect for Postgres 10 and later.
  */
 public class PostgreSQL10Dialect extends PostgreSQL95Dialect {
 
+	public PostgreSQL10Dialect() {
+		registerFunction( "timezone", new StandardSQLFunction( "timezone", StandardBasicTypes.TIMESTAMP ) );
+	}
+
```

Comment:
I think `timezone` function is included in postgreSQL from the very beginning. E.g. https://www.postgresql.org/docs/8/functions-datetime.html